### PR TITLE
Batter Up!

### PR DESCRIPTION
**Pull request by FalseIncarnate**
### Tweaks how baseball bats work to be a little more interesting.

Previously, baseball bats were guaranteed to knock back on every hit with no cooldown between knockbacks.

Now, baseball bats don't knock back on a normal hit. Bunt your rival into the ground and keep him there!

But wait, what's this? You enjoyed swinging for the fences with your rival's face as your new tee-ball?!

You still can! Simply take up your swinging stance and build up a single, double, or even triple power hit! Charging your swing this way grants increasing amounts of knockback for the next hit within a short
time, but you have to stand still to build up the swing, and the power will quickly drop off after a few seconds after you stop charging your swing, so time it right to avoid a strike-out!

### How it works:
Activating the baseball bat in hand will begin to charge up a swing, with increasing tiers of power being obtained with longer charge times.
- Tier 0 (basic attacks) swings don't knock back anymore, but also don't require any wind-up.
- Tier 1 charged swings grant a 1-tile knockback, and take 1 second to charge up.
- Tier 2 charged swings increase the knockback to 2 tiles, and requires an additional 2 seconds charge time
- Tier 3 charged swings bring the knockback to 3 tiles, but at the cost of an additional 4 seconds of wind-up.
  - For reference, baseball bats would previously knockback 1 or 2 tiles at random. A tier 3 swing is thus more knockback than previously obtainable with a normal baseball bat.

When you begin charging, you will automatically progress through each tier, meaning to go from a basic hit to a tier 3 swing will take 7 seconds of charging. Moving, attacking, dropping your bat, etc. during this charge-up will interrupt the charge at whichever tier was last achieved.

Once you stop charging, your charged swing will begin to rapidly decay back to a normal hit after 4 seconds. During this delay and subsequent delay, you can move and make a single swing at the current tier as normal. Dropping your bat will also instantly reset the charged swing to 0, so keep a good grip!

Every time you gain a tier of swing charge, there will be a small accompanying tone. When your swing fully decays back to a normal hit, there is also a different tone with a chat message to inform you that
you've lost the swing.

The homerun bat still functions the same as before, except losing the knockback on the basic hits. As before, this is still an admin item.

:cl:
tweak: Baseball bats now require you wind-up your swing before sending your victim sailing into the outfield. Swing for the fences!
/:cl:

### CODE REVIEW

**Comment by Birdtalon** [comment]
Very nice idea but you could probably reduce those timers by half. One second is a lot in combat and I think the key here is to just stop knockback spam.

**Comment by uraniummeltdown** [comment]
It's interesting but I think this should be made into another item (metal baseball bat?) while leaving the wooden one the same

**Comment by alexkar598** [comment]
I find it pretty useless,a part of this game combat is moving a lot or you will get killed. Now you need to stand in the same location for 1 sec or more to have a better weapon?

**Comment by KasparoVy** [comment]
Goodbye knickback spam.

**Comment by FalseIncarnate** [comment]
In my testing, the tier 1 and tier 2 charge timers are really quite quick, I actually added the audio cues so I could reliably test the different tiers. In actual combat this will be a tiny bit more pronounced, but not enough to warrant lowering the times without seeing it in action yet.

One of my main goals with this change was to make the weapon more interesting and require some decision-making by the wielder. As the baseball bat already does damage comparable to a toolbox, it isn't a weapon to scoff at, and the ability to knock an enemy out of melee range is a mechanic not many other weapons possess. By making the player consciously activate the knockback, this makes the weapon more versatile while still keeping one of its defining traits intact. Now you don't have to constantly chase your victim while you dole out some street justice after every hit unless you choose to charge up a swing beforehand, but can now decide to commit to the swing and send them farther than before at the cost of a temporary vulnerability.

I frankly wouldn't be surprised to see people actually using the charge to their advantage to lure their opponent in closer, only to send them reeling with a mid-tier swing.

Remember, when you charge, you will hit with whichever tier you last achieved. This means if you charge for 6 seconds, you will pop out a double (which you attained at 3 seconds), but at the 7th second you open up a triple. You can still swing while charging, it just ends and unleashes the charged swing.

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -163,29 +163,77 @@ obj/item/weapon/wirerod/attackby(obj/item/I, mob/user, params)
...
+	if(swing_power == 1  && at_bat)
+		if(do_after(user, 20, target = src) && at_bat)
+			swing_power = 2
+			user.playsound_local(null, 'sound/items/timer.ogg', 50, 0)
+	if(swing_power == 2  && at_bat)
+		if(do_after(user, 40, target = src) && at_bat)
+			swing_power = 3
+			user.playsound_local(null, 'sound/items/timer.ogg', 50, 0)
+	at_bat = FALSE
+	spawn(30)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/weaponry.dm]: Don't use spawns; we have an `addtimer` system for this now.

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -163,29 +163,77 @@ obj/item/weapon/wirerod/attackby(obj/item/I, mob/user, params)
...
 	to_chat(user, "<span class='warning'>You begin gathering strength...</span>")
 	playsound(get_turf(src), 'sound/magic/lightning_chargeup.ogg', 65, 1)
 	if(do_after(user, 90, target = src))
 		to_chat(user, "<span class='userdanger'>You gather power! Time for a home run!</span>")
 		homerun_ready = 1
+
+/obj/item/weapon/melee/baseball_bat/proc/wind_down(mob/user)
+	if(at_bat || !swing_power)
+		return
+	spawn(10)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/weaponry.dm]: Timers.

**Comment by Fox-McCloud** [comment]
>It's interesting but I think this should be made into another item (metal/plasteel baseball bat?) while leaving the wooden one the same

This is exactly what I said earlier.

I don't really see any major problems with bats, ergo I don't see a need to change them; no design problem is being solved--this is changing them for the sake of a new mechanic--in my opinion, that belongs better on a new weapon than an old one.

How this will actually play out in combat, I don't think will be what is intended; it'll almost never be advantageous to hit them with a charged up hit (except for as an opening move, whereby it'll be obvious you're doing it), vs just bashing them one more time with the bat.

I get what's being aimed for here, I just don't think it'll accomplish it.

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (FalseIncarnate) [reply code/game/objects/items/weapons/weaponry.dm]: Not arguing against using the timers, but for the sake of information, wouldn't a timer be more expensive in this case as you'd be adding additional proc calls?

Comment (Fox-McCloud) [reply code/game/objects/items/weapons/weaponry.dm]: `spawn` is pretty expensive, in and of itself; it's more than just a single proc call.

**Comment by taukausanake** [comment]
I'd go with making the wooden bat be this one and the metal bat (I'd vote made of plasteel) be the insta home run one. Wood isn't too difficult to get, you just tear down the barricades in maint or assembly line

**Comment by FalseIncarnate** [comment]
Homerun bat is and will remain the admin spawned item. It not only sends the victim flying, but also calls ex_act on them, so it can blow off a limb in the process

**Comment by FalseIncarnate** [comment]
spawns replaced with timers, also clarified a minor mistake in the original description: your charged swing loses strength after a total of 4 seconds. It took 3 seconds plus an additional 1 second for the first reduction of tier, but these were calculated by 2 separate spawns due to how the code was ordered. I just reordered it to keep this same timing but be cleaner codewise

**Comment by FalseIncarnate** [comment]
Don’t feel like continuing to pursue this change, someone else can scavenge this into a new item if they want, but it’s not a big enough deal for me to try pushing for this change.

Was an idea that ultimately wasn’t needed for any real reason other than my own opinions on design.

**Pull request closed without merging**